providers/hubspot/record.go

```diff
@@ -0,0 +1,55 @@
+package hubspot
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+
+	"github.com/amp-labs/connectors/common"
+)
+
+/*
+   docs:
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/companies
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/contacts
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/deals
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/tickets
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/line_items
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/products
+*/
+
+//nolint:gochecknoglobals
+var getRecordSupportedObjectsToPathMap = map[string]string{
+	"company":   "companies",
```

Comment:
Instead of a map this can be a set and URL part could be created using `naming.NewPluralString(objectName).String()`.

What about the variable name: `supportedObjectsByWebhook`.

Reply:
The `handy` package (soon `datautils`) has: 
```
supported := handy.NewSet("calls", "orders", "customers")
if !supported.Has("orders") {
 return Err..
}
```
But that is syntactic sugar underneath it is `map[string]struct{}`.